TypeCobol/Compiler/Domain/Validator/SymbolValidator.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.Parser.Generated;
+using TypeCobol.Compiler.Symbols;
+
+namespace TypeCobol.Compiler.Domain.Validator
+{
+    /// <summary>
+    /// Symbol Validator.
+    /// </summary>
+    public class SymbolValidator : Symbol.AbstractSymbolVisitor<bool, Object>
+    {
+        /// <summary>
+        /// List of unvalidated symbol
+        /// </summary>
+        public List<Symbol> Unvalidated { get; set; }
+
+        /// <summary>
+        /// The TypeValidator instance that is used.
+        /// </summary>
+        public TypeValidator TypValidator { get; internal set; }
+
+        /// <summary>
+        /// Constructor based on a TypeValidator.
+        /// </summary>
+        public SymbolValidator() : this(new TypeValidator())
+        {
+            TypValidator.SymValidator = this;
+        }
+
+        /// <summary>
+        /// Constructor with a TypeValidator as argument.
+        /// </summary>
+        /// <param name="typeValidator">The TypeValidator instance to be used when validating types</param>
+        public SymbolValidator(TypeValidator typeValidator)
+        {
+            Unvalidated = new List<Symbol>();
+            TypValidator = typeValidator ?? new TypeValidator(this);
+        }
+
+        /// <summary>
+        /// Default implementation we only check that the type isn't null.
+        /// </summary>
+        /// <param name="s">The symbol to check</param>
+        /// <param name="arg">Any argument</param>
+        /// <returns></returns>
+        public override bool VisitSymbol(Symbol s, object arg)
+        {
+            if (s.Type == null)
+            {
+                Unvalidated.Add(s);
+                return false;
+            }
+            else if (!s.Type.Accept(TypValidator, arg))
+            {
+                Unvalidated.Add(s);
+                return false;
+            }
```

Comment:
Can be simplified as
`
if (s.Type == null || !s.Type.Accept(TypValidator, arg))
{
    Unvalidated.Add(s);
    return false;
}
`

Reply:
ok